File: backend/src/main/kotlin/com/petshop/models/Coupon.kt
package com.petshop.models

import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalDateTime
import java.util.*

data class Coupon(
    val id: UUID = UUID.randomUUID(),
    val storeId: UUID? = null,
    val code: String,
    val title: String,
    val description: String? = null,
    val type: String, // "discount", "freeship", "store"
    val discountAmount: BigDecimal,
    val minSpend: BigDecimal = BigDecimal.ZERO,
    val expiryDate: LocalDate,
    val conditions: List<String> = emptyList(),
    val color: String? = null,
    val badgeIconUrl: String? = null,
    val createdAt: LocalDateTime = LocalDateTime.now()
)

File: frontend/lib/coupon.ts
export type CouponType = "discount" | "freeship" | "store";

export type CouponStatus = "available" | "collected" | "used" | "expired";

export interface Coupon {
  id: string;
  code: string;
  title: string;
  description?: string;
  discountAmount: number; // จำนวนส่วนลด (บาท)
  minSpend: number; // ขั้นต่ำในการใช้ (บาท)
  type: CouponType;
  status: CouponStatus;
  expiryDate: string; // ISO date string
  conditions: string[];
  // Store-specific coupons
  storeId?: string;
  storeName?: string;
  storeLogo?: string;
  color?: "red" | "orange" | "green";
  badgeIcon?: string;
}

export interface UserCoupon extends Coupon {
  collectedAt?: string; // ISO date string
  usedAt?: string; // ISO date string
}

export interface CouponState {
  allCoupons: Coupon[]; // All available coupons in the system
  userCoupons: UserCoupon[]; // User's collected coupons
}
